### PR TITLE
[Merged by Bors] - refactor: use junk values in Submodule.toLinearPMap

### PR DESCRIPTION
**Pull request by mcdoll**
Change the definition of `Submodule.toLinearPMap` to use a junk value in the case that the condition that the subspace defines the graph of a function is not satisfied. In this case we define `Submodule.toLinearPMap` as the zero map. The domain is the same so that the domain does not depend on the graph-condition.

---

the file is not completely ported, but since I wrote the relevant parts not too long ago I am very confident that it is not used in the unported files. I am also fine with waiting until all downstream dependencies are ported.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/LinearPMap.lean [hunk 1]
```diff
@@ -945,41 +945,59 @@ theorem valFromGraph_mem {g : Submodule R (E × F)}
...
+  map_smul' := fun a v => by
+    have hsmul := (g.map (LinearMap.fst R E F)).smul_mem a v.2
+    have hav := valFromGraph_mem hg hsmul
+    have hav' := g.smul_mem a (valFromGraph_mem hg v.2)
+    rw [Prod.smul_mk] at hav'
+    exact (existsUnique_from_graph @hg hsmul).unique hav hav'
+
+open Classical
+
 /-- Define a `LinearPMap` from its graph. -/
```
Comment (sgouezel) [inline Mathlib/LinearAlgebra/LinearPMap.lean:970]: Can you expand the docstring to mention the junk value?

**Review by sgouezel — COMMENTED** [review]
LGTM, thanks.
bors d+

**Comment by sgouezel** [comment]
The title of the PR is not very clear. `refactor: use junk values in Submodule.toLinearPMap`, maybe? Can you also expand the PR description?

**Comment by mcdoll** [comment]
bors merge

**Pull request closed without merging**